sklearn/impute/_iterative.py

```diff
@@ -565,9 +569,6 @@ def fit_transform(self, X, y=None):
 
         self.imputation_sequence_ = []
 
-        self._min_value = -np.inf if self.min_value is None else self.min_value
```

Comment:
Keep these lines.

Reply:
Done - but those lines are actually not needed as the use of `as_float_array` will convert `None` to `np.nan` anyways, and we later convert `np.nan` to -/+ `np.inf`. Your suggestion of using `check_array` doesn't work with integer arrays when we later try to convert `np.nan` in those arrays to `np.inf` - and we need to do this last step because the user can give an array-like input with `np.nan` or `None` in it which the condition above will miss. Actually, the condition above fails even for scalar `np.nan`.

Anyways, I have written up a detailed explanation for all that I tried below, have a look at that. The code could be more parsimonious, but I think we need a separate pull request to do that.